pkg/dns/ibm/private/dnssvcs_provider.go

```diff
@@ -0,0 +1,273 @@
+package private
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	dnssvcsv1 "github.com/IBM/networking-go-sdk/dnssvcsv1"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	dnsclient "github.com/openshift/cluster-ingress-operator/pkg/dns/ibm/private/client"
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	iov1 "github.com/openshift/api/operatoringress/v1"
+)
+
+var (
+	_                       dns.Provider = &Provider{}
+	log                                  = logf.Logger.WithName("dns")
+	defaultDNSSVCSRecordTTL              = int64(120)
+)
+
+type Provider struct {
+	dnsService dnsclient.DnsClient
+	config     Config
+}
+
+type Config struct {
```

Comment:
I think that makes sense to me. I see `InstanceID` is a bit overloaded since it could be different types of ID's for public or private (CRN ID vs. IstanceID?). But I think this is okay.